File: cpp/cpp_1000-/1184.cc
// C++
// leetcode 1184
// https://leetcode.cn/problems/distance-between-bus-stops/

class Solution {
public:
    int distanceBetweenBusStops(vector<int>& distance, int start, int destination) {
    	if (start == destination)
    		return 0;
    	if (start > destination)
    		swap(start, destination);

    	int sum = 0, cur = 0;
    	for (int i = 0; i < distance.size(); i++) {
    		sum += distance[i];
    		if (i >= start && i < destination)
    			cur += distance[i];
    	}
    	return min(cur, sum - cur);
    }
};

File: Go/Go_1000-/1184.go
// Go
// leetcode 1184
// https://leetcode.cn/problems/distance-between-bus-stops/

func distanceBetweenBusStops(distance []int, start int, destination int) int {
	if start == destination {
		return 0
	}

	if start > destination {
		start, destination = destination, start
	}

	sum, cur := 0, 0
	for i, d := range distance {
		sum += d
		if i >= start && i < destination {
			cur += d
		}
	}
	if cur * 2 > sum {
		cur = sum - cur
	}

	return cur
}